app/lib/get-user-ribbon-data.js

```diff
@@ -1,36 +1,37 @@
 import getAllProjectPreferences from './get-all-project-preferences';
 import getColorFromString from './get-color-from-string';
+import apiClient from 'panoptes-client/lib/api-client'
 
 function getUserRibbonData(user) {
   return getAllProjectPreferences(user).then((prefsResources) => {
     const awaitProjects = Promise.all(prefsResources.map((prefsResource) => {
-      return prefsResource.get('project').catch(() => {
+      return apiClient.type('projects').get(prefsResource.links.project).catch(() => {
         return null;
       });
     }));
 
-    const awaitOwners = awaitProjects.then((projects) => {
+    const awaitSubjectSets = awaitProjects.then((projects) => {
       return Promise.all(projects.map((project) => {
-        if (project === null) {
+        return project.get('subject_sets').catch(() => {
           return null;
-        } else {
-          return project.get('owner').catch(() => {
-            return null;
-          });
-        }
+        });
       }));
     });
 
-    return Promise.all([awaitProjects, awaitOwners]).then(([projects, owners]) => {
+    return Promise.all([awaitProjects, awaitSubjectSets]).then(([projects, subject_sets]) => {
       return prefsResources.map((prefsResource, i) => {
-        if (projects[i] !== null && owners[i] !== null) {
+        const sortedSubjects = subject_sets[i].sort((a, b) => {
```

Comment:
Is the goal to only sort the first page of subject sets? There will be some projects that will have 20+ subject sets (Notes from Nature I think? Gravity Spy will too eventually). I would suggest passing in the sort param into the API call instead of doing this.